infrastructure/utils.py

```diff
@@ -156,7 +162,11 @@ def create_expenditure(project, finance_phase, amount):
 def create_finance_phase(s):
     phase, year = parse_finance_phase(s)
     fy, _ = models.FinancialYear.objects.get_or_create(budget_year=year)
-    phase, _ = models.BudgetPhase.objects.get_or_create(name=phase)
+
+    try:
+        phase = models.BudgetPhase.objects.get(name=phase)
+    except BudgetPhase.DoesNotExist as e:
+        raise ValueError("Could not find an existing budget phase matching those supplied, no phase created")
```

Comment:
```suggestion
        raise ValueError("Could not find an existing budget phase matching those supplied")
```

In 30 years no one will know this code once created phases. Especially if we fix the method name to be "get_budget_phase" rather than "create_budget_phase" :)